docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,369 @@
+
+However, the introduction of the `ShootedSeed` resource would require adapting all `gardener-controller-manager` controllers that operate on Shoots (for example, shoot hibernation and maintenance controllers) to also operate on ShootedSeeds, since the Shoot resource created as a result of reconciling a ShootedSeed is owned by the latter and should never be changed directly, since any changes to its `spec` would be immediately overwritten by the reconciliation. Instead, changes should be applied to the `shoot` section of the ShootedSeed, which would result in propagating them to the owned Shoot by the shooted seed controller.
+
+### ShootedSeedSets
+
+Similarly to a [ReplicaSet](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/), the purpose of a ShootedSeedSet is to maintain a stable set of replica [ShootedSeeds](#shootedseeds) available at any given time. As such, it is used to guarantee the availability of a specified number of identical ShootedSeeds.
+
+#### ShootedSeedSet Resource
+
+The `ShootedSeedSet` resource has a `selector` field that specifies how to identify ShootedSeeds it can acquire, a number of `replicas` indicating how many ShootedSeeds it should be maintaining, and a shooted seed `template` specifying the data of new ShootedSeeds it should create to meet the number of replicas criteria. A ShootedSeedSet then fulfills its purpose by creating and deleting ShootedSeeds as needed to reach the desired number. 
+
+A ShootedSeedSet is linked to its ShootedSeeds via the ShootedSeeds' `metadata.ownerReferences` field, which specifies what resource the current object is owned by. All ShootedSeeds acquired by a ShootedSeedSet have their owning ShootedSeedSet's identifying information within their `ownerReferences` field. 
+
+Example `ShootedSeedSet` resource:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: ShootedSeedSet
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      foo: bar
+  template:
+    metadata:
+      labels:
+        foo: bar
+    spec: # ShootedSeed resource
+      shoot: 
+        ...
+      seed: 
+        ...
+```
+
+#### ShootedSeedSet Controller
+
+ShootedSeedSets are reconciled by a new *shooted seed set controller* in `gardener-controller-manager`. During the reconciliation this controller creates and deletes ShootedSeeds in response to changes to the `replicas` and `selector` fields.
+
+**Note:** The introduction of the `ShootedSeedSet` resource would not require any changes to `gardenlet` or to existing `gardener-controller-manager` controllers. 
+
+#### ShootedSeedDeployments
```

Comment:
Although this is quite far ahead, I'm not sure, if rolling Seeds like deployments will be the right approach.
I can imagine, that rolling `ShootedSeedSet`s rather like `StatefulSet`s will be more like what we desire (rolling seed by seed, stopping on rollout-failures ref https://github.com/gardener/gardener/issues/87 and so on).

As this is also quite vague, I think it makes sense to just mention both approaches here for now and then decide later on once we have a more clear picture.

Reply:
Yes, agreed, will adapt the document accordingly.

Reply:
Done